core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java

```diff
@@ -52,11 +46,16 @@ public JavaDoubleRDD union(JavaDoubleRDD... rdds) {
     if (rdds.length == 0) {
       throw new IllegalArgumentException("Union called on empty list");
     }
-    List<JavaPairRDD<K, V>> rest = new ArrayList<>(rdds.length - 1);
+    rest = populateRDDList(rdds);
+    return union(rdds[0], rest);
+  }
+  
+  public static final List<T> populateRDDList(T rdds) {
```

Comment:
I agree with making the method "private".But rdds is not any varargs so why do we need ...??